### ISSUE TITLE
Ansible-playbook fails to run due to SSL Error

### ISSUE THREAD

**Issue by brett-fitz** [splunk/docker-splunk#300]
I am trying to start my splunk docker container in an environment which utilizes TLS decrypt. 
This container fails to start because the ansible-playbook requires SSL verification and this fails because there is no option to provide a ca-bundle at runtime. Is there any known workarounds?

```
ansible-playbook 2.9.1
  config file = /opt/ansible/ansible.cfg
  configured module search path = [u'/opt/ansible/library', u'/opt/ansible/apps/library', u'/opt/ansible/ansible_commands']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.16 (default, Oct 10 2019, 22:02:15) [GCC 8.3.0]
Traceback (most recent call last):
  File "inventory/environ.py", line 450, in <module>
    main()
  File "inventory/environ.py", line 436, in main
    getSplunkInventory(inventory)
  File "inventory/environ.py", line 84, in getSplunkInventory
    inventory["all"]["vars"] = getDefaultVars()
  File "inventory/environ.py", line 150, in getDefaultVars
    getSplunkApps(defaultVars)
  File "inventory/environ.py", line 166, in getSplunkApps
    data={"username": splunkbase_username, "password": splunkbase_password})
  File "/usr/lib/python2.7/dist-packages/requests/api.py", line 116, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='splunkbase.splunk.com', port=443): Max retries exceeded with url: /api/account:login/ (Caused by SSLError(SSLError(1, u'[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)'),))
splunkd.pid file is unreadable.
```

**Comment by arctan5x** [comment]
@brett-fitz 

Hi. What are the arguments (or a yaml) to start your container?

**Comment by brett-fitz** [comment]
Hey man! Thanks for getting back to me, here is my docker-compose.yml
```
version: "3.7"
services:
  splunkd:
    image: splunk/splunk:7.3
    container_name: splunk
    env_file: .env
    environment:
      - SPLUNK_START_ARGS=--accept-license
      - SPLUNK_PASSWORD=$SPLUNK_PASSWORD
      - SPLUNKBASE_USERNAME=$SPLUNKBASE_USERNAME
      - SPLUNKBASE_PASSWORD=$SPLUNKBASE_PASSWORD
      - SPLUNK_APPS_URL=https://splunkbase.splunk.com/app/2686/release/3.2.0/download,https://splunkbase.splunk.com/app/3411/release/2.7.5/download
      - JAVA_VERSION=openjdk:8
      - DEBUG=true
    ports:
      - 8000:8000
      - 8089:8089
    volumes:
      - splunk-var:/opt/splunk/var
      - splunk-etc:/opt/splunk/etc
```

Note (unrelated): I have to use `7.3` not `latest` because openjdk:8 (required by db_connect) will only successfully install on `7.3`

**Comment by brett-fitz** [comment]
Note: The issue of not being able to install JDK8 was due to persistent volumes. For anyone experiencing this issue:

```
docker-compose down   # Removes container + Network
docker volume prune      # Removes unused volumes (be careful!) 
docker-compose up        
```

I tested splunk:8.0.1 and was successful finally. 

Best workaround for the above issue (SSL Errors) is to whitelist *.splunk.com. Furthermore, if you install Java, you will have to whitelist the domain that is used for acquiring the GPG key for Java, as well as any redhat domains used for mirrors.
